tests/bouncer/util_test.py

```diff
@@ -16,184 +16,96 @@ def test_parse_document_raises_if_annotated_deleted():
-                "selector": [{}]
-            }],
-            "group": "__world__",
-            "shared": True,
-            "text": ""
-        }
-    })["text"]
-
+def test_parse_document_returns_boilerplate_when_no_text(es_annotation_doc):
+    text = util.parse_document(es_annotation_doc)["text"]
     assert text == util.ANNOTATION_BOILERPLATE_TEXT
 
 
-def test_parse_document_returns_show_metadata_true_when_shared_and_world():
-    show_metadata = util.parse_document({
-        "_id": "annotation_id",
-        "_source": {
-            "target": [{
-                "source": "http://example.com/example.html",
-                "selector": [{}]
-            }],
-            "group": "__world__",
-            "shared": True
-        }
-    })["show_metadata"]
-
+def test_parse_document_returns_show_metadata_true_when_shared_and_world(es_annotation_doc):
+    show_metadata = util.parse_document(es_annotation_doc)["show_metadata"]
     assert show_metadata is True
 
 
-def test_parse_document_returns_document_uri_from_web_uri_when_pdf():
-    document_uri = util.parse_document({
-        "_id": "annotation_id",
-        "_source": {
-            "target": [{
-                "source": "urn:x-pdf:the-fingerprint",
-                "selector": []
-            }],
-            "group": "__world__",
-            "shared": True,
-            "document": {"web_uri": "http://example.com/foo.pdf"}
-        }
-    })["document_uri"]
+def test_parse_document_returns_document_uri_from_web_uri_when_pdf(es_annotation_doc):
+    es_annotation_doc["_source"]["target"][0]["source"] = "urn:x-pdf:the-fingerprint"
+    es_annotation_doc["_source"]["document"] = {"web_uri": "http://example.com/foo.pdf"}
+
+    document_uri = util.parse_document(es_annotation_doc)["document_uri"]
 
     assert document_uri == "http://example.com/foo.pdf"
 
 
-def test_parse_document_raises_when_uri_from_web_uri_not_string_for_pdfs():
+def test_parse_document_raises_when_uri_from_web_uri_not_string_for_pdfs(es_annotation_doc):
+    es_annotation_doc["_source"]["target"][0]["source"] = 52
+    es_annotation_doc["_source"]["document"] = {"web_uri": "http://example.com/foo.pdf"}
+
     with pytest.raises(util.InvalidAnnotationError) as exc:
-        util.parse_document({
-            "_id": "annotation_id",
+        util.parse_document(es_annotation_doc)
+
+    assert exc.value.reason == "uri_not_a_string"
+
+
+@pytest.fixture
+def es_annotation_doc():
+    """
+    Minimal JSON document for an annotation as returned from Elasticsearch.
+
+    This contains only fields which can be assumed to exist on all annotations.
+    """
+    return {"_id": "annotation_id",
             "_source": {
                 "target": [{
-                            "source": "urn:x-pdf:the-fingerprint",
-                            "selector": []
-                            }],
+                    "source": "http://example.com/example.html",
+                    "selector": [{}]
```

Comment:
Some of the original test fixtures have this one-element list containing an empty dictionary, others have an empty list. Does that make a difference in these test cases?

Reply:
It doesn't make a difference given the way the logic is written but it _could_, so I have 1) removed the "selector" field here since not all annotations will have a selector and 2) added a test case with a non-quote selector which is more realistic since I'm pretty sure we will have those in ES.